src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -50,14 +50,31 @@ public boolean supports(Class<?> clazz) {
 
   public static class SchemaValidationContext {
 
+    enum Operation {
+      CREATE("schema"),
+      UPDATE("changes");
+
+      private final String fieldName;
+
+      Operation(String fieldName) {
+        this.fieldName = fieldName;
+      }
+    }
```

Comment:
Using an enum like this is OK (and better than using a boolean or making the caller provide the field name directly) but I'd consider another approach -- using a factory method. It would look something like:
```
  public static class SchemaValidationContext {

    private final Map<String, HashSet<String>> tableColumnMap = new HashMap<>();
    private final Map<String, HashSet<String>> tableArrayColumns = new HashMap<>();
    private final Set<String> relationshipNameSet = new HashSet<>();
    private final String fieldName;

    SchemaValidationContext(String fieldName) {
      this.fieldName = fieldName;
    }

    static SchemaValidationContext forUpdate() {
      return new SchemaValidationContext("changes");
    }

    static SchemaValidationContext forCreate() {
      return new SchemaValidationContext("schema");
    }
...
  private void validateSchema(DatasetSpecificationModel schema, Errors errors) {
    SchemaValidationContext context = SchemaValidationContext.forCreate();
```